smarts/core/argoverse_map.py

```diff
@@ -909,6 +904,11 @@ def query(
                     return [path[: (lookahead + 1)] for path in hit]
                 return None
 
+    @cached_property
+    def _lanepoints(self):
+        assert self._map_spec.lanepoint_spacing > 0
+        return LanePoints.from_argoverse(self, spacing=self._map_spec.lanepoint_spacing)
```

Comment:
This slightly helps but generally not. The issue I was thinking of is that all waypoints are generated on this line whenever that might be. It will not scale.

Reply:
It helps for the case where we load the roadmap without "using" it, which I believe happens in scenario build.